StardewArchipelago/GameModifications/CodeInjections/Modded/JunimoShopInjections.cs

```diff
@@ -0,0 +1,224 @@
+            _stardewItemManager = stardewItemManager;
+            _junimoShopGenerator = junimoShopGenerator;
+        }
+
+
+
+        private static ShopMenu _lastShopMenuUpdated = null;
+        
+        // public override void update(GameTime time)
+        public static bool Update_JunimoWoodsAPShop_Prefix(ShopMenu __instance, GameTime time)
+        {
+            try
+            {
+                // We only run this once for each menu
+                if (__instance.storeContext != "Custom_JunimoWoods")
+                {
+                    return true;
+                }
+                var firstItemForSale = "";
+                if (__instance.forSale.Count == 0) // in case the shop is emptied on a second pass
+                {
+                    firstItemForSale = "";
+                }
+                else
+                {
+                    firstItemForSale = __instance.forSale.First().Name; // Fighting CP moment
+                }
+                if (!_junimoFirstItems.Contains(firstItemForSale) && _lastShopMenuUpdated == __instance)
+                {
+                    return true;
+                }
+                _lastShopMenuUpdated = __instance;
+                var color = _firstItemToColor[firstItemForSale];
+                if (color == "Purple")
+                {
+                    var purpleJunimo = __instance.portraitPerson;
+                    __instance.exitThisMenuNoSound();
+                    DaJunimo = purpleJunimo;
+                    DaJunimo.Name = "Purple Junimo";
+                    DaJunimo.displayName = "Purple Junimo";
+                    PurpleJunimoSpecialShop();
+                    return true;
+                }
+                __instance.itemPriceAndStock = _junimoShopGenerator.GetJunimoShopStock(color, __instance.itemPriceAndStock);
+                __instance.forSale = __instance.itemPriceAndStock.Keys.ToList();
+                __instance.potraitPersonDialogue = _junimoPhrase[color];
+                return true; //  run original logic
+            }
+            catch (Exception ex)
+            {
+                _monitor.Log($"Failed in {nameof(Update_JunimoWoodsAPShop_Prefix)}:\n{ex}", LogLevel.Error);
+                return true; // run original logic
+            }
+        }
+
+                private class PurpleJunimo
+        {
+            public StardewItem OfferedItem {get; set;}
+            public int Amount {get; set;}
+        }
+
+        public static void PurpleJunimoSpecialShop()
+        {
+            var junimoWoods = Game1.player.currentLocation;
+            var friendsMet = Game1.player.friendshipData.Count();
+            var fakeStock = new Dictionary<string, int>(){
+                {"Money", 10000},
+                {"Dewdrop", 2500},
+                {"Friendship", 200 * friendsMet}
+            };
+            PurpleJunimoOptions = new Dictionary<string, PurpleJunimo>();
+            var randomSeed = 0;
```

Comment:
You don't need to do that. Just using the same Random instance and doing Next() over and over will get new random numbers in the same seeded formula